pkg/doc/verifiable/credential.go

```diff
@@ -219,7 +219,21 @@ type embeddedCompositeIssuer struct {
 type CredentialDecoder func(dataJSON []byte, credential *Credential) error
 
 // CredentialTemplate defines a factory method to create new Credential template.
-type CredentialTemplate func() *Credential
+// It's possible that Verifiable Credential could be not relevant to the Template's data model.
+// It can be determined by Template checking the types of the Verifiable Credential.
+// In this case error is returned.
+type CredentialTemplate func(credentialTypes []string) (*Credential, error)
```

Comment:
It is incorrect to determine the credential based just on a token encountered in the `type` field. You also need the `@context`.

Types are defined within `@context`s. The `VerifiableCredential` type is defined in the "https://www.w3.org/2018/credentials/v1" context (you can fetch that URL to verify this yourself). Likewise, additional types need to be defined in a context somewhere, and that includes adding it to the `@context` property.

Reply:
Hey @llorllale ! I agree with you that we need to pass `@context` as well as `type` so that we can avoid potential collision of same types defined in different contexts. I will change this. Thank you!